aspnetcore/release-notes/aspnetcore-3.0.md

```diff
@@ -0,0 +1,477 @@
+        {
+            configHost.SetBasePath(Directory.GetCurrentDirectory());
+            configHost.AddJsonFile("hostsettings.json", optional: true);
+            configHost.AddEnvironmentVariables(prefix: "PREFIX_");
+            configHost.AddCommandLine(args);
+    });
+```
+
+Prior to the release of ASP.NET Core 3.0, environment variables prefixed with `ASPNETCORE_` were loaded for host configuration of the Web Host. In 3.0, `AddEnvironmentVariables` is used to load environment variables prefixed with `DOTNET_` for host configuration with `CreateDefaultBuilder`. 
+
+To configure host options in code, call `Configure<HostOptions>` on the host builder's services provided by `ConfigureServices`:
+
+```csharp
+public static IHostBuilder CreateHostBuilder2(string[] args) =>
+    Host.CreateDefaultBuilder(args)
+        .ConfigureServices((hostContext, services) =>
+        {
+            services.Configure(option =>
+            {
+                option.ShutdownTimeout = System.TimeSpan.FromSeconds(20);
+            });
+        });
+```
+
+For more information, see the following articles:
+
+* <xref:fundamentals/configuration/index>
+* <xref:fundamentals/host/generic-host>
+
+## Kestrel
+
+* Kestrel configuration has been updated for the migration to the Generic Host. In 3.0, Kestrel is configured on the web host builder provided by `ConfigureWebHostDefaults`.
+* Connection Adapters have been removed from Kestrel and replaced with Connection Middleware, which is similar to HTTP Middleware in the ASP.NET Core pipeline but for lower-level connections.
+* The Kestrel transport layer has been exposed as a public interface in `Connections.Abstractions`.
+* Ambiguity between headers and trailers has been resolved by moving trailing headers to a new collection.
+* Synchronous IO APIs, such as `HttpReqeuest.Body.Read`, are a common source of thread starvation leading to app crashes. In 3.0, `AllowSynchronousIO` is disabled by default.
+
+For more information, see <xref:migration/22-to-30>.
+
+## Hosting EventSourceProvider new request counters
+
+The Hosting EventSourceProvider (Microsoft.AspNetCore.Hosting) emits the following request counters:
+
+* `requests-per-second`
+* `total-requests`
+* `current-requests`
+* `failed-requests`
+
+## Endpoint routing
+
+See https://github.com/aspnet/AspNetCore.Docs/issues/14291
+
+## Health Checks
+
+Health Checks use endpoint routing with the Generic Host. In `Startup.Configure`, call `MapHealthChecks` on the endpoint builder with the endpoint URL or relative path:
+
+```csharp
+app.UseEndpoints(endpoints =>
+{
+    endpoints.MapHealthChecks("/health");
+});
+```
+
+Health Checks endpoints can:
+
+* Specify one or more permitted hosts/ports.
+* Require authorization.
+* Require CORS.
+
+For more information, see the following articles:
+
+* <xref:migration/22-to-30#health-checks>
+* <xref:host-and-deploy/health-checks>
+
+## Pipes on HttpContext
+
+TODO by PU
```

Comment:
LGTM.

Reply:
This is fine with me too. It might be worth calling out these are analogues to the HttpRequest/Response.Body Streams, but I could go either way on that.